### PR TITLE
Update node-fetch definitions to v2.1.2

### PR DESCRIPTION
**Pull request by nikcorg**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/bitinn/node-fetch/blob/master/CHANGELOG.md#v200
- [x] Increase the version number in the header if appropriate.
- [ ] ~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~

### CODE REVIEW

**Review by torstenwerner — APPROVED** [review]

File: types/node-fetch/index.d.ts [hunk 1]
```diff
@@ -1,110 +1,174 @@
-// Type definitions for node-fetch v1.6
+// Type definitions for node-fetch v2.1.2
```
Comment (DanielRosenwasser) [inline types/node-fetch/index.d.ts]: Looks like our lint rules are disabled for this package. Can you enable the lints in our `tslint.json` and fix up the version string to only be a `MAJOR.MINOR` version (e.g. `4.7`), as opposed to including the patch or containing a leading `v` (e.g. `v1.2.3`).

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/node-fetch/index.d.ts (hunk 1 above)
Comment (nikcorg) [reply types/node-fetch/index.d.ts]: @DanielRosenwasser thanks for taking the time to review. I've enabled the linting rules and addressed the shortcomings.

**Comment by nikcorg** [comment]
@DanielRosenwasser I hate to be pushy, but if you could find the time to review the changes on this PR, I'd be very grateful.

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**